File: surface/widgets/watchkit/components/widgets/complex-inputs/RangeSlider.swift
// ============================================================
// Clef Surface WatchKit Widget — RangeSlider
//
// watchOS SwiftUI implementation. Simplified for compact
// round-screen display with minimal interaction patterns.
// ============================================================

import SwiftUI

struct RangeSliderView: View {
    @Binding var lowerValue: Double; @Binding var upperValue: Double; var range: ClosedRange<Double> = 0...100
    var body: some View { VStack { Slider(value: $lowerValue, in: range); Slider(value: $upperValue, in: range); Text("\(Int(lowerValue)) - \(Int(upperValue))").font(.caption2) } }
}

File: surface/widgets/wear-compose/components/widgets/complex-inputs/RangeSlider.kt
// ============================================================
// Clef Surface Wear Compose Widget - RangeSlider
//
// Wear OS Compose implementation. Adapted for round screen
// using ScalingLazyColumn, Chip, ToggleChip, TimeText.
// ============================================================

package clef.surface.wearcompose.components.widgets.complexinputs

import androidx.compose.runtime.*
import androidx.compose.foundation.layout.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.Alignment
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.wear.compose.material.*
import androidx.wear.compose.foundation.lazy.*

@Composable
fun RangeSlider(
    lowerValue: Float = 0f,
    upperValue: Float = 100f,
    onLowerChange: (Float) -> Unit = {},
    onUpperChange: (Float) -> Unit = {},
    range: ClosedFloatingPointRange<Float> = 0f..100f,
    modifier: Modifier = Modifier
) {
    Column(modifier = modifier) {
        Text("Min: ${lowerValue.toInt()}", fontSize = 10.sp)
        InlineSlider(value = lowerValue, onValueChange = onLowerChange, valueRange = range, steps = 9, modifier = Modifier.fillMaxWidth())
        Text("Max: ${upperValue.toInt()}", fontSize = 10.sp)
        InlineSlider(value = upperValue, onValueChange = onUpperChange, valueRange = range, steps = 9, modifier = Modifier.fillMaxWidth())
    }
}
